client/src/main/java/io/pravega/client/batch/BatchClient.java

```diff
@@ -51,18 +51,21 @@
     <T> SegmentIterator<T> readSegment(Segment segment, Serializer<T> deserializer);
 
     /**
-     * Provides a SegmentIterator to read the events after the startingOffset but before the
-     * endingOffset in the requested segment.
+     * Provides a SegmentIterator to read the events after the startingOffset in the requested
+     * segment ending at the current end of the segment.
      * 
-     * Offsets can be obtained by calling {@link SegmentIterator#getOffset()} or {@link SegmentInfo#getLength()}
+     * Offsets can be obtained by calling {@link SegmentIterator#getOffset()} or
+     * {@link SegmentInfo#getLength()}. There is no validation that the provided offset actually
```

Comment:
`SegmentInfo` does not contain a length attribute. Perhaps you are referring to `writeOffset`? In any case, please review this sentence as it does not seem to be accurate.